### PR TITLE
572 adex vignette

### PR DESCRIPTION
**Pull request by kabis-ops**
Hi @Roche-GSK/admiral-devs  - the adex vignette should be updated now. Please let me know any comment. @millerg23 - may I please ask you to check this is aligned with the updated adex script?
Thanks

### CODE REVIEW

**Comment by millerg23** [comment]
Hi @kabis-ops
Apologies for late response. In the script we add the ADSL vars after the initial creation of EXDOSE and EXADJ (and adding on SUPP). We also store the ADSL vars in a vector.
```
 # Get list of ADSL vars required for derivations
adsl_vars <- vars(TRTSDT, TRTSDTM, TRTEDTM)

# Part 1
# Join ADSL with ex and derive required dates, variables
adex0 <- ex %>%
  # Join ADSL with EX (only ADSL vars required for derivations)
  left_join(
    adsl %>% select(STUDYID, USUBJID, !!!adsl_vars),
    by = c("STUDYID", "USUBJID")
  ) %>%
```

Then at the end add all other ADSL vars:
```
# Join all ADSL with EX
adex <- adex %>%

  left_join(select(adsl, !!!admiral:::negate_vars(adsl_vars)),
            by = c("STUDYID", "USUBJID")
  )
```

Only other thing I see is ASEQ is slightly different, in script its:
```
# Calculate ASEQ
  derive_obs_number(
    new_var = ASEQ,
    by_vars = vars(STUDYID, USUBJID),
    order = vars(PARCAT1, ASTDT, VISIT, VISITNUM, EXSEQ, PARAMN),
    check_type = "error"
  )
```
I don't think this really matters.

**Comment by kabis-ops** [comment]
@millerg23 , thanks for the comment. I have updated the vignette.
One thing I wonder, @thomas-neitmann - is it expected that we have to specify `admiral:::` to call the `negate_vars()` function? and this generates a warning regarding attributes.

**Comment by kabis-ops** [comment]
+ `filter_if`

**Comment by thomas-neitmann** [comment]
@kabis-ops @bundfussr Agree, let's export these functions.

**Comment by bundfussr** [comment]
@thomas-neitmann , should we do it in this PR or should we create a new issue?

File: vignettes/bds_exposure.Rmd [hunk 1]
```diff
@@ -1,8 +1,7 @@
 ---
 title: "Creating an Exposure (BDS) ADaM"
```
Comment (bundfussr) [inline vignettes/bds_exposure.Rmd:2]: Should we rename it to "Creating a BDS Exposure ADaM" to be in line with the other BDS vignettes?

File: vignettes/bds_exposure.Rmd [hunk 2]
```diff
@@ -125,16 +133,21 @@ Example calls:
...
+                 display_vars = vars(USUBJID, VISIT, EXSTDTC , EXENDTC ,ASTDT, AENDT))
+```
+
+
+```{r eval=TRUE, echo=FALSE}
 adex <- select(adex, -ASTDTF, -AENDTF)
 ```
 
 
 This next examples demonstrates the datetime imputation features available
```
Comment (bundfussr) [inline vignettes/bds_exposure.Rmd:149]: I think "This next examples demonstrates" is not correct. What about "The next example demonstrates"?

I also wonder if there should be a comma before "where", i.e., 'in the `derive_vars_dtm()` function, where the time is imputed as "00:00:00"'.

**Review by bundfussr — CHANGES_REQUESTED** [review]

**Comment by thomas-neitmann** [comment]
Let's create a separate issue for that.

File: vignettes/bds_exposure.Rmd (hunk 1 above)
Comment (kabis-ops) [reply vignettes/bds_exposure.Rmd:2]: OK, updated

File: vignettes/bds_exposure.Rmd (hunk 2 above)
Comment (kabis-ops) [reply vignettes/bds_exposure.Rmd:149]: ok, updated

**Comment by kabis-ops** [comment]
@bundfussr - updated as per your comments
@thomas-neitmann - i still have the call to `admiral:::`, this will need to be updated once the functions are exported.

**Comment by thomas-neitmann** [comment]
@kabis-ops I just opened a PR for exporting these utilities. Once that it merged let's merged `devel` into this branch to get the changes and remove the `admiral:::` prefix.

**Comment by kabis-ops** [comment]
Ok, PR approved :)

**Comment by thomas-neitmann** [comment]
Thanks, merged 😉

**Comment by kabis-ops** [comment]
@thomas-neitmann : updated

**Comment by thomas-neitmann** [comment]
@bundfussr Could you approve? Otherwise it's blocked as you had requested changes.

**Review by bundfussr — APPROVED** [review]

**Pull request merged**